gdal/frmts/hdf5/bagdataset.cpp

```diff
@@ -471,8 +480,21 @@ GDALDataset *BAGDataset::Open( GDALOpenInfo *poOpenInfo )
     }
 
     // Open the file as an HDF5 file.
+#ifdef ENABLE_UFFD
```

Comment:
this common logic repeated 3 times could be factored in a single function

Reply:
Addressed.